locations/spiders/cvs.py

```diff
@@ -0,0 +1,111 @@
+import scrapy
+import re
+from locations.items import GeojsonPointItem
+
+DAY_MAPPING = {'M':'Mo', 'T': 'Tu', 'W': 'We', 'F': 'Fr', 'Sat': 'Sa', 'Sun':'Su'}
+
+
+class CVSSpider(scrapy.Spider):
+
+    name = "cvs"
+    allowed_domains = ["www.cvs.com"]
+    download_delay = 1.5
+    start_urls = (
+        'https://www.cvs.com/store-locator/cvs-pharmacy-locations',
+    )
+
+    def parse_day(self, day):
+
+        if re.search('Sat', day) or re.search('Sun', day):
+            return DAY_MAPPING[day.strip()]
+
+        if re.search('-', day):
+            days = day.split('-')
+            osm_days = []
+            if len(days) == 2:
+                for day in days:
+                    osm_day = DAY_MAPPING[day.strip()]
+                    osm_days.append(osm_day)
+            return "-".join(osm_days)
+
+    def parse_times(self, times):
+        hours_to = [x.strip() for x in times.split('-')]
+        cleaned_times = []
+
+        for hour in hours_to:
+            if re.search('PM$', hour):
+                hour = re.sub('PM', '', hour).strip()
+                hour_min = hour.split(":")
+                if int(hour_min[0]) < 12:
+                    hour_min[0] = str(12 + int(hour_min[0]))
+                cleaned_times.append(":".join(hour_min))
+
+            if re.search('AM$', hour):
+                hour = re.sub('AM', '', hour).strip()
+                hour_min = hour.split(":")
+                if len(hour_min[0]) <2:
+                    hour_min[0] = hour_min[0].zfill(2)
+                else:
+                    hour_min[0] = str(12 + int(hour_min[0]))
+
+                cleaned_times.append(":".join(hour_min))
+        return "-".join(cleaned_times)
+
+    def parse_hours(self, lis):
+
+        hours = []
+        for li in lis:
+            day = li.xpath('normalize-space(.//span[@class="day single-day"]/text() | .//span[@class="day"]/text())').extract_first()
+            times = li.xpath('.//span[@class="timings"]/text()').extract_first()
+            if times and day:
+                parsed_time = self.parse_times(times)
+                parsed_day = self.parse_day(day)
+                hours.append(parsed_day + ' ' + parsed_time)
+
+        return ";".join(hours) or 'No Store Hours'
```

Comment:
Please leave out the `opening_hours` property if there are no hours to include instead of including the string "No Store Hours".

Reply:
👍